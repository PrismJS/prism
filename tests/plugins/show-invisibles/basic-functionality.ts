import { createTestSuite } from '../../helper/prism-dom-util';


describe('Show Invisibles', () => {
	const { it } = createTestSuite({
		languages: 'javascript',
		plugins: 'show-invisibles'
	});

	it('should show invisible characters', ({ util }) => {
		util.assert.highlightElement({
			language: 'javascript',
			code: `  \t\n\r\n\t\t`,
			format: false
		});
	});

	it('should show invisible characters inside tokens', ({ util }) => {
		util.assert.highlightElement({
			language: 'javascript',
			code: `/* \n */`,
			format: false
		});
	});
});
