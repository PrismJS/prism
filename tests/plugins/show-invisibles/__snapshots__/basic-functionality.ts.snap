// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show Invisibles should show invisible characters 1`] = `
"<span class=\\"token space\\"> </span><span class=\\"token space\\"> </span><span class=\\"token tab\\">	</span><span class=\\"token lf\\">
</span><span class=\\"token crlf\\">
</span><span class=\\"token tab\\">	</span><span class=\\"token tab\\">	</span>"
`;

exports[`Show Invisibles should show invisible characters inside tokens 1`] = `
"<span class=\\"token comment\\">/*<span class=\\"token space\\"> </span><span class=\\"token lf\\">
</span><span class=\\"token space\\"> </span>*/</span>"
`;
